File: dmn-core/src/test/resources/tck/cl2/expected/dmn/0010-multi-output-u/type/TApprovalImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tApproval"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TApprovalImpl implements TApproval {
        private String status;
        private String rate;

    public TApprovalImpl() {
    }

    public TApprovalImpl(String rate, String status) {
        this.setRate(rate);
        this.setStatus(status);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Rate")
    public String getRate() {
        return this.rate;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Rate")
    public void setRate(String rate) {
        this.rate = rate;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Status")
    public String getStatus() {
        return this.status;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Status")
    public void setStatus(String status) {
        this.status = status;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-core/src/test/resources/tck/cl2/expected/kotlin/dmn/0010-multi-output-u/type/TApprovalImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tApproval"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TApprovalImpl : TApproval {
    @get:com.fasterxml.jackson.annotation.JsonGetter("Status")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Status")
    override var status: String? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Rate")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Rate")
    override var rate: String? = null

    constructor() {
    }

    constructor (rate: String?, status: String?) {
        this.rate = rate
        this.status = status
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
